arangod/V8Server/v8-voccursor.cpp

```diff
@@ -102,6 +102,10 @@ static void JS_CreateCursor(v8::FunctionCallbackInfo<v8::Value> const& args) {
     arangodb::Cursor* cursor =
         cursors->createFromQueryResult(std::move(result),
                                        static_cast<size_t>(batchSize), ttl, true);
+    if (cursor == nullptr) {
+      TRI_V8_THROW_EXCEPTION_FULL(TRI_ERROR_SHUTTING_DOWN,
+           "coordinator soft shutdown, new cursors blocked");
+    }
```

Comment:
As mentioned somewhere above, I would remove the check here and make `CursorRepository::createFromQueryResult()` throw an appropriate exception when we are in soft shutdown.

Reply:
I vaguely remember that at some stage we wanted to get away from exceptions for error handling and rely more on return values, but my memory might not serve me well. Anyway, I do not really care...

Reply:
Done.